### ISSUE TITLE
Do we need to provide `db.Close()` to release resources

### ISSUE THREAD

**Issue by lonre** [VinGarcia/ksql#18]
Hi

Do we need to provide `db.Close()` to release resources

```Go
func (db *DB) Close() error
```

**Comment by VinGarcia** [comment]
Hi @lonre thanks for creating the first issue =D

So yeah we'll need that, I was thinking that we should support both: adapters that implement the `io.Closer` interface and adapters that don't, so we could maybe create a function like this:

```
func (c DB) Close() error {
    closer, ok := c.db.(io.Closer)
    if ok {
        return closer.Close()
     }
     return nil
}
```

We would then have to update each adapter to include this method.

If you want to implement it, it would be great, otherwise, I can do it maybe this weekend or on the next.

**Comment by lonre** [comment]
Hi @VinGarcia 

`ksql.Tx` implements `ksql.DBAdapter`, if `ksql.DBAdapter` implements `io.closer`, do you have any idea for `ksql.Tx`?

**Comment by VinGarcia** [comment]
So, I was thinking of keeping these interfaces separate, so I would not change ksql.DBAdapter mostly because this interface represents the minimum set of operations that I need to make it work. And I don't need the Close() method for that.

Meaning that if eventually, we get an adapter that does not have a Close() operation we can work in the same way we already do for the existing ones.

So the example Close function I wrote above would use DuckTyping to check whether that particular adapter was "closeable", if not we would do nothing and everything would keep working.

So, I hope I got your question right, if not tell me.
